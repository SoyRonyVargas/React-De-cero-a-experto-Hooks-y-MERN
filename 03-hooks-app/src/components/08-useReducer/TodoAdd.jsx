import React from 'react'

const TodoAdd = ({ handleSubmit , ref }) => {
    return (
        <div className='col'>
            <h3>
                Agregar Todo
            </h3>
            <form onSubmit={handleSubmit}>
                <div class="mb-3">
                    <textarea
                        className="form-control"
                        rows={3}
                        defaultValue={""}
                        ref={ref}
                    />
                </div>

                <button type='submit' className='btn btn-primary btn-small btn-sm'>
                    Agregar
                </button>
            </form>
        </div>
    )
}

export default TodoAdd