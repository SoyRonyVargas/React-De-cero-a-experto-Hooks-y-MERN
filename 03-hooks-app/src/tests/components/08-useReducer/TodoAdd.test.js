import TodoAdd from "../../../components/08-useReducer/TodoAdd"
import { shallow } from "enzyme"

describe('Probando el componente <TodoAdd/>', () => { 

    const payload = {
        handleSubmit: jest.fn(),
        ref: null
    }

    const wrapper = shallow(
        <TodoAdd
            {...payload}
        />
    )

    test('deberia mostrarse correctamente', () => { 

        expect(wrapper).toMatchSnapshot()

    })

    test('NO deberia enviar el formulario', () => { 

        const formSubmit = wrapper.find("form").prop("onSubmit")

        formSubmit(({ preventDefault(){} }))

        expect( payload.handleSubmit ).toHaveBeenCalledTimes(1)

    })

})