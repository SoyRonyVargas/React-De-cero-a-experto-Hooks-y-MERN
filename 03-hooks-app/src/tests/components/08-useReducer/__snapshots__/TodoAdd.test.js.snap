// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <TodoAdd/> deberia mostrarse correctamente 1`] = `
<div
  className="col"
>
  <h3>
    Agregar Todo
  </h3>
  <form>
    <div
      class="mb-3"
    >
      <textarea
        className="form-control"
        defaultValue=""
        rows={3}
      />
    </div>
    <button
      className="btn btn-primary btn-small btn-sm"
      type="submit"
    >
      Agregar
    </button>
  </form>
</div>
`;
